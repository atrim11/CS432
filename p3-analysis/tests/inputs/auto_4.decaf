def int multiply(int x, int y) {
    return x * y;
}

def int sumArray(int arr, int size) {
    int sum;
    int i;
    sum = 0;
    i = 0;
    while (i < size) {
        sum = sum + arr[i];
        i = i + 1;
    }
    return sum;
}

def int recursiveSum(int n) {
    return n + recursiveSum(n - 1);  // Error: no base case for recursion
}
int numbers[3];
int otherNumbers[5];
def void main() {
    // Nested function calls with mismatches
    int result;
    int total;
    int index;
    int finalResult;
    int x;
    result = multiply(multiply(5, 2), "text");  // Error: wrong type for inner multiply call

    // Array size mismatch
    numbers[4] = 10;  // Error: out of bounds

    // Recursion test
    result = recursiveSum(5);  // Infinite recursion due to missing base case

    // Using undeclared variables
total = sumArray(numbers, size);  // Error: 'size' is undeclared

    // Multiple array accesses
    otherNumbers[-1] = 100;  // Error: negative array index
    index = otherNumbers[10];  // Error: out-of-bounds array access

    // Multiple control structure errors
    if (result + 10) {  // Error: non-boolean in 'if'
        print_str("Control test");
    }

    while (result) {  // Valid case but could lead to infinite loop
        result = result - 1;
    }

    // Invalid usage of void functions in expressions
    x = printMessage("Hello") + 5;  // Error: void used in expression

    // Function call with too few parameters
    result = multiply(5);  // Error: multiply expects 2 parameters

    // Type mismatch in return
    finalResult = printMessage("Done");  // Error: void function assigned to int
}
