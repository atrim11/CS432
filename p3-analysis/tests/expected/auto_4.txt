Non-array 'arr' accessed as an array on line 11
Type mismatch in parameter 1 of call to 'multiply': expected int but found str on line 29
Array 'numbers' accessed without index on line 38
Symbol 'size' undefined on line 38
Type mismatch in parameter 1 of call to 'sumArray': expected int but found ??? on line 38
Type mismatch: bool expected but int found on line 45
Type mismatch: bool expected but int found on line 49
Symbol 'printMessage' undefined on line 54
Invalid number of function arguments on line 57
Symbol 'printMessage' undefined on line 60
'main' must return an integer
